src/session_impl.cpp

```diff
@@ -1701,6 +1702,20 @@ namespace aux {
 		reopen_listen_sockets();
 	}
 
+	namespace
+	{
+		struct listen_endpoint_t
+		{
+			listen_endpoint_t(address adr, int p, std::string dev, bool s)
+				: addr(adr), port(p), device(dev), ssl(s) {}
```

Comment:
Not important. None of this code is remotely performance sensitive. For most people it won't be executed more than a few times a day.